### ISSUE TITLE
False positive `inert_defer` with condition compilation

### ISSUE THREAD

**Issue by yuri-qualtie** [realm/SwiftLint#4615]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Rule `inert_defer` is false positive when defer is wrapped in condition compilation. For example: `#if os(macOS)`

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint Test.swift

Linting Swift files at paths Test.swift
Linting 'Test.swift' (1/1)
.../Test.swift:6:9: warning: Inert Defer Violation: If defer is at the end of its parent scope, it will be executed right where it is anyway. (inert_defer)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version - `0.50.1`
* Installation method used  - Homebrew
* Paste your configuration file - default 

* Xcode version - `Xcode 14.0.1 Build version 14A400`
* Run - `swiftlint lint --path Test.swift --no-cache --enable-all-rules`
```
../Test.swift:4:5: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly. (explicit_acl)
../Test.swift:6:9: warning: Inert Defer Violation: If defer is at the end of its parent scope, it will be executed right where it is anyway. (inert_defer)
../Test.swift:3:8: warning: Missing Docs Violation: public declarations should be documented. (missing_docs)
../Test.swift:3:8: warning: Required Deinit Violation: Classes should have an explicit deinit method. (required_deinit)
```

```swift
public class Test {
    func hello(name: String) -> String {
        #if os(macOS)
        // this trigger violation: Inert Defer Violation: If defer is at the end of its parent scope, it will be executed right where it is anyway. (inert_defer)
        defer {
            print(self)
        }
        #endif

        print(name)

        return "Hello " + name
    }
}
```

**Comment by stasswtf** [comment]
Also false positive:

```swift
let task = Task {
	#if DEBUG
		let stopwatch = Stopwatch()
		defer { Logger.default.debug("\(name ?? "Tool"): \(stopwatch).") }
	#endif
	// …
}
```

**Comment by SimplyDanny** [comment]
The rule seems to be obsolete now given that the compiler warns already:
![Screenshot 2022-12-04 at 09 22 43](https://user-images.githubusercontent.com/16365760/205481185-acbd98df-1111-4054-a044-3683c9dd20fc.png)

**Comment by SimplyDanny** [comment]
The the rule will be considered deprecated in the next release. Nevertheless, the bug has also been fixed.
